### PR TITLE
fix(developer): kmcmplib was clobbering previous output 🗜

### LINKED ISSUE #9125: bug(developer): kmc-kmn appears to go wrong after building multiple times in one session 🗜
I've patched up kmc here to run the build 100 times in a row. After 18 builds, we get a failure like follows:

```
$ node --enable-source-maps /c/Projects/keyman/app/developer/src/kmc build release/gff/gff_amharic
...
<snipped>
...
INFO 5002: Building release/gff/gff_amharic
INFO 5002: Building C:\Projects\keyman\keyboards\release\gff\gff_amharic\source\gff_amharic.kmn
WARN 20209f: The touch layout font should be the same for all platforms.
INFO 5006: gff_amharic.kmn built successfully.
INFO 5002: Building C:\Projects\keyman\keyboards\release\gff\gff_amharic\source\gff_amharic.kps
Buffer(0) [Uint8Array] []
FATAL 405001: Unexpected exception: RangeError: Offset is outside the bounds of the DataView

Call stack:
RangeError: Offset is outside the bounds of the DataView
    at DataView.getUint32 (<anonymous>)
    at DecodeStream.<computed> [as readUInt32LE] (file:///C:/Projects/keyman/app/common/web/types/node_modules/restructure/src/DecodeStream.js:79:35)
    at NumberT.decode (file:///C:/Projects/keyman/app/common/web/types/node_modules/restructure/src/Number.js:20:36)
    at Struct._parseFields (file:///C:/Projects/keyman/app/common/web/types/node_modules/restructure/src/Struct.js:41:20)
    at Struct.decode (file:///C:/Projects/keyman/app/common/web/types/node_modules/restructure/src/Struct.js:12:10)
    at Struct.fromBuffer (file:///C:/Projects/keyman/app/common/web/types/node_modules/restructure/src/Base.js:7:17)
    at KmxFileReader.read (C:\Projects\keyman\app\common\web\types\src\kmx\kmx-file-reader.ts:57:40)
    at PackageVersionValidation.getKeyboardFileData (C:\Projects\keyman\app\developer\src\kmc-package\src\compiler\package-version-validation.ts:156:23)
    at PackageVersionValidation.validateAndUpdateVersions (C:\Projects\keyman\app\developer\src\kmc-package\src\compiler\package-version-validation.ts:44:25)
    at KmpCompiler.transformKpsToKmpObject (C:\Projects\keyman\app\developer\src\kmc-package\src\compiler\kmp-compiler.ts:145:26)
```

I originally observed this when batch compiling all keyboards in the keyboards repository, with the following command (noting that xargs breaks the find result into batches of some unspecified size):

```shell
find release -name '*.kpj' | grep -v 'vietnamese_' | xargs sh -c 'node --enable-source-maps /c/Projects/keyman/app/developer/src/kmc build $* || exit 255'
```

So I don't think this is specific to gff_amharic, but it's a nice big keyboard which probably triggers failure quickly.

Notes:
* Hardcoded path to debug build of kmc here; you could shorten to `kmc build $*` in production.
* The `|| exit 255` pattern is to force xargs to die on _any_ error, because xargs will only die on exit code 255.
* Excluding `vietnamese_*` because of memory and performance issues for this test; that's being tracked in #9112.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9125.

Running kmc on a set of keyboards would sporadically fail after some time. It turns out this was related to memory allocation, specifically, resizing the WASM module's memory buffer caused `TypedArray` views into the memory to be reset!

By default, when a `Uint8Array` is created from an `ArrayBuffer` (e.g. `Module.HEAP8.buffer`), it is a dynamic view into that buffer. This module buffer can be dynamically reallocated at any time, which can happen when allocating memory in WASM code (so the change will look _really_ weird in a stack trace). Thus, to ensure we don't trip over ourselves, we need to copy the buffer. Fortunately, creating a `Uint8Array` from a `Uint8Array` copies the data, and is pretty quick.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-kmn/src/compiler/compiler.ts [hunk 1]
```diff
@@ -213,6 +213,22 @@ export class KmnCompiler implements UnicodeSetParser {
...
+   * allocating memory in WASM code (so the change will  look _really_ weird in
+   * a stack trace). Thus, to ensure we don't trip over ourselves, we need to
+   * copy the buffer. Fortunately, creating a `Uint8Array` from a `Uint8Array`
+   * copies the data, and is pretty quick.
+   * @param offset    Offset into the WASM memory space, in bytes
+   * @param size      Size of the buffer to copy, in bytes
+   * @returns         A _copy_ of the data in a new Uint8Array
+   */
+  private copyWasmBuffer(offset: number, size: number): Uint8Array {
+    return new Uint8Array(new Uint8Array(Module.HEAP8.buffer, offset, size));
```
Comment (jahorton) [inline developer/src/kmc-kmn/src/compiler/compiler.ts:229]: Do we not need to `delete[]` the intermediate `Uint8Array`?

Comment (mcdurdin) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: It's Javascript, there ain't no `delete` :grin:

Comment (jahorton) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: Oh, right, this one's TS as opposed to the other PRs in the group, which are C++-based.

**Review by jahorton — APPROVED** [review]

File: developer/src/kmc-kmn/src/compiler/compiler.ts (hunk 1 above)
Comment (srl295) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: bikeshed: i might have named it `dupWasmBuffer` or `cloneWasmBuffer` or something. `copy` to me sounds like strcpy

Comment (mcdurdin) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: `copy` to me sounded like `memcpy` :-P

Comment (jahorton) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: `cpy` that.

Comment (srl295) [reply developer/src/kmc-kmn/src/compiler/compiler.ts:229]: if memcpy took `NULL` as a first argument yeah, meaning "surprise me!" :D

**Pull request merged**